src/osd/PrimaryLogPG.cc

```diff
@@ -3472,6 +3474,29 @@ void PrimaryLogPG::dec_refcount(ObjectContextRef obc, const object_ref_delta_t&
   }
 }
 
+void PrimaryLogPG::dec_refcount_by_dirty(OpContext* ctx, const object_manifest_t& updated_chunks)
+{
+  object_ref_delta_t refs;
+  ObjectContextRef cobc = nullptr;
+  ObjectContextRef obc = ctx->obc;
+  // Look over previous snapshot, then figure out whether updated chunk needs to be deleted
+  if (obc->ssc->snapset.clones.size()) {
+    cobc = get_prev_clone_obc(obc);
+  }
+  updated_chunks.calc_refs_to_drop_on_modify(
+    cobc ? &cobc->obs.oi.manifest : nullptr,
+    refs);
+  if (!refs.is_empty()) {
+    hobject_t soid = obc->obs.oi.soid;
+    ceph_assert(ctx);
+    ctx->register_on_commit(
+      [soid, this, refs](){
+	ObjectContextRef obc = get_object_context(soid, false, NULL);
+	ceph_assert(obc);
+	dec_refcount(obc, refs);
```

Comment:
Why does dec_refcount take an ObjectContext?  It simply calls refcount_manifest which doesn't need an actual object ObjectContext.  I suppose I must have missed this in a previous PR, but that would eliminate a metadata read in the post-commit path, so I think you should go ahead and do it in this PR (though in a separate commit).